lib/client.js

```diff
@@ -247,1672 +245,1670 @@ function nop (error) {
  * @param {Object} [options] - connection options
  * (see Connection#connect)
  */
-function MqttClient (streamBuilder, options) {
-  let k
-  const that = this
+class MqttClient extends EventEmitter {
+  constructor (streamBuilder, options) {
+    super()
 
-  if (!(this instanceof MqttClient)) {
-    return new MqttClient(streamBuilder, options)
-  }
+    let k
+    const that = this
 
-  this.options = options || {}
+    this.options = options || {}
 
-  // Defaults
-  for (k in defaultConnectOptions) {
-    if (typeof this.options[k] === 'undefined') {
-      this.options[k] = defaultConnectOptions[k]
-    } else {
-      this.options[k] = options[k]
+    // Defaults
+    for (k in defaultConnectOptions) {
+      if (typeof this.options[k] === 'undefined') {
+        this.options[k] = defaultConnectOptions[k]
+      } else {
+        this.options[k] = options[k]
+      }
     }
```

Comment:
This might be out of scope for this PR, but it looks like this code can be simplified with spread syntax